test/hotspot/jtreg/compiler/arguments/TestMethodArguments.java

```diff
@@ -0,0 +1,160 @@
+
+                        @Check(test = "$test")
+                        public static void $check(#type res) {
+                            if (res != $golden[$nextInput]) {
+                                throw new RuntimeException("wrong result " + res + "!=" + $golden[$nextInput]);
+                            }
+                            $nextInput = ($nextInput + 1) % $inputs.length;
+                        }
+                        """));
+    }
+
+    public static String generate(CompileFramework comp) {
+        List<Object> tests = new LinkedList<>();
+        for (int i = MIN; i <= MAX; ++i) {
+            tests.add(generateTest(CodeGenerationDataNameType.ints(), i).asToken());
+            tests.add(generateTest(CodeGenerationDataNameType.floats(), i).asToken());
+            // Longs and doubles take up double as much space in the parameter list as other
+            // primitive types (e.g., int). We therefore have to divide by two to fill up
+            // the same amount of space as for ints and floats.
+            tests.add(generateTest(CodeGenerationDataNameType.longs(), i / 2).asToken());
+            tests.add(generateTest(CodeGenerationDataNameType.doubles(), i / 2).asToken());
+        }
+        return Template.make(() -> Template.body(
+                Template.let("classpath", comp.getEscapedClassPathOfCompiledClasses()),
+                """
+                        import java.util.Arrays;
+                        import java.util.stream.*;
+                        import compiler.lib.generators.*;
+                        import compiler.lib.ir_framework.*;
+                        import compiler.lib.template_framework.library.*;
+
+                        public class InnerTest {
+
+                            static RestrictableGenerator<Integer> generatorInteger = Generators.G.uniformInts();
+                            static RestrictableGenerator<Long> generatorLong = Generators.G.uniformLongs();
+                            static RestrictableGenerator<Float> generatorFloat = Generators.G.uniformFloats();
+                            static RestrictableGenerator<Double> generatorDouble = Generators.G.uniformDoubles();
+
+                            public static void main() {
+                                TestFramework framework = new TestFramework(InnerTest.class);
+                                framework.addFlags("-classpath", "#classpath");
+                                framework.start();
+                            }
+
+                            public static <T> T[][] generateInput(Generator<T> t, T[][] array) {
+                                for (int i = 0; i < array.length; i++) {
+                                    for (int j = 0; j < array[i].length; j++) {
+                                        array[i][j] = t.next();
+                                    }
+                                }
+                                return array;
+                            }
```

Comment:
Seems like we need to add some convenience "fill" methods to the template library. We'll get there eventually, just keep this for now.